spec/models/membership_spec.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe Membership, type: :model do
+  describe 'basic validations' do
+    it { is_expected.to belong_to(:user) }
+    it { is_expected.to belong_to(:group) }
+    it { is_expected.to validate_presence_of(:user) }
+    it { is_expected.to validate_presence_of(:group) }
+  end
+
+  describe 'group draw and user draw must match' do
+    it do
+      user = FactoryGirl.create(:student_in_draw)
+      group = FactoryGirl.create(:open_group)
+      membership = FactoryGirl.build(:membership, user: user, group: group)
+      expect(membership.valid?).to be_falsey
+    end
+  end
+
+  describe 'cannot change group' do
+    it do
+      group = FactoryGirl.create(:full_group)
+      membership = group.memberships.last
+      new_leader = FactoryGirl.create(:student, draw: group.draw)
+      membership.group = FactoryGirl.create(:open_group, leader: new_leader)
```

Comment:
this highlights some ugliness from how our `Group` factory is set up: we can't override the draw by itself because we end up with a mismatched draw between the group and the leader.

I tried to fix it but couldn't get anything working; passing optional attributes to association factories isn't easy.